### ISSUE TITLE
Can't get table width to fit to page

### ISSUE THREAD

**Issue by johnvanbreda** [eKoopmans/html2pdf#44]
Hi,
I've got a page with an HTML table on it, set to 100% width on the page. I'd like to output the table into a PDF which fits the table to the width of the page and the document will create as many pages as is necessary for the whole table. My code is as follows:

```
html2pdf($('div.node-content')[0], {
      filename:     'myReport.pdf',
      margin:       0,
      image:        { type: 'jpeg', quality: 0.98 },
      html2canvas:  { dpi: 192, letterRendering: true },
      jsPDF:        { format: 'a4', orientation: 'p' }
});
```

I've found that the table does not fit the PDF page width and a couple of columns on the right of the table are clipped off. I've experimented with the format option in the jsPDF setting as well as the margin option but I can't seem to affect this behaviour.

Any thoughts on how to do this would be much appreciated.
John

**Comment by johnvanbreda** [comment]
A partial answer to my question - it seems that the table will scale to fit 100% width up to a certain size. If the table is larger than the width available then the scaling "gives up" and doesn't shrink to fit, it just clips the rightmost columns.

**Comment by eKoopmans** [comment]
Hey @johnvanbreda, really sorry I never got back to you. I've been swamped. So did you sort out the issue? What you've described (table filling the page width, and producing as many pages as necessary for all the rows) is exactly what html2pdf should produce. You're right that it can cause problems if the table's content exceeds its container's size.

**Comment by johnvanbreda** [comment]
Hi @eKoopmans, thanks for getting back to me. Yes, it seems the problem relates to the table content exceeding the container size - I was expecting this to scale the content either to increase or decrease it to fit, but it seems it will only scale up not down.
I have a hack which works OK, but it is a bit messy. Basically I look at the $(table).width() value for each table and if it is over a certain width (1100px for landscape for example), I apply a CSS attribute to decrease the font size by the same ratio by which the content is too wide. I'm not convinced it is a perfect solution but it is "good enough".

**Comment by eKoopmans** [comment]
Right, I hear you. There have been a few requests for "single-page PDFs", where the entire content would be shrunk down to the page size - this might fall into that category. Or more generally, a "fit-to-width" option (which could still span multiple pages).

Currently html2pdf just restricts the page size and lets the content re-flow naturally to fit in the space. It works well for many circumstances, but it sounds like it's not perfect for yours. I'll add this to the project tracking for the single-page PDFs, might be a while until I get to it though!

**Comment by johnvanbreda** [comment]
Yes - the fit-to-width option would meet my needs. Thanks.

**Comment by enyosolutions** [comment]
Hello ! 

i'm having the same problem !

**Comment by vietnguyen3107** [comment]
Hi, Me too. Can you release new version with the fit-to-width option as soon as possible, please!

**Comment by eKoopmans** [comment]
Hi everyone, I haven't forgotten about this, but time is short these days. I hope I'll be able to get to this soon.

**Comment by MartinHlavna** [comment]
Hello, 
Affects me as well

**Comment by phyzalis** [comment]
Same probleme here :)

**Comment by edupression** [comment]
same prob here! :)

**Comment by Quanda** [comment]
Same here

**Comment by dorsaffer** [comment]
same here

**Comment by Lakshamana** [comment]
any updates with this issue?

**Comment by ahmedmgh67** [comment]
Same here

**Comment by KZeni** [comment]
Just in case it helps some people, I ended up switching over to using PrintThis (https://github.com/jasonday/printThis) where it isolates content and triggers the browser's print command which then allows the visitor to save to PDF. This then doesn't suffer this issue (this issue then being important enough for me to switch over to this alternative method). I can definitely see how this then doesn't meet the needs of others since dealing with the print dialogue would be ideal to avoid (I'd like to avoid it as well, but I found the other approach sufficient for now.)

**Comment by neldreth2021** [comment]
Anyone mind giving me an example for fit-to-width? Is this a CSS option I need to include in my container styling, or a configuration option for underlying jsPDF?

**Comment by aGitForEveryone** [comment]
Is there any update on this topic? The image is still not resized to fit the width of the specified jsPDF format (A4 in my case). It fits the height of 2 pages exactly though, but the width overflows.

**Comment by aGitForEveryone** [comment]
I found a solution for this problem. You need to set the `width` property in `html2canvas`. Find out the width of your element in pixels and pass that to `html2canvas`, something like:
```
html2canvas:  { width: element_width_in_pixels },
```

**Comment by IngridCosovan** [comment]
Hello! This is what worked for me is (but the solution is feasible because I don't need it portrait, and the format is just for view mode, when you actually print it you select A4):
jsPDF: {orientation: 'landscape', format: 'a3'}

**Comment by KGESH** [comment]
> Hello! This is what worked for me is (but the solution is feasible because I don't need it portrait, and the format is just for view mode, when you actually print it you select A4): jsPDF: {orientation: 'landscape', format: 'a3'}

Thanks! it's worked for me. 

`const option = {
            margin: 1,
            image: {type: 'jpeg', quality: 0.98},
            html2canvas: {scale: 2},
            pagebreak: {mode: 'avoid-all'},
            jsPDF: {unit: 'mm', format: 'a3', orientation: 'p'}

        };

        html2pdf().set(option).from(source).toPdf().save('my.pdf');`

**Comment by YangW0223** [comment]
me too

**Comment by vfa-tanna** [comment]
this work with me! 🆘

```javascript
const info = document.getElementsByClassName('xxxx')[0] as HTMLElement;
info.style.width = info.scrollWidth.toString() + 'px';
```

**Comment by rizahoemae** [comment]
hi guys i already testing this out. at first my pdf look like this 
![image](https://github.com/eKoopmans/html2pdf.js/assets/55048401/9e4b3447-351e-46df-876f-5b45d9de6003)
after trial error i found this solution how to make my image fit to width and height propotional as i capture so it look like this 
![image](https://github.com/eKoopmans/html2pdf.js/assets/55048401/e5305e03-1b12-433e-a444-e42b2d78dd36)
i followed this [tips from stackoverflow](https://stackoverflow.com/questions/36472094/how-to-set-image-to-fit-width-of-the-page-using-jspdf#:~:text=If%20you%20need%20to%20get%20width%20100%25%20of%20PDF%20file%20and%20auto%20height%20you%20can%20use%20getImageProperties%20property%20of%20html2canvas%20library). hope this helps

**Comment by linshenjianyue** [comment]
> I found a solution for this problem. You need to set the `width` property in `html2canvas`. Find out the width of your element in pixels and pass that to `html2canvas`, something like:
> 
> ```
> html2canvas:  { width: element_width_in_pixels },
> ```

same problem.
using this way dont solution my prob, pdf cut right column and narrow.
maybe go back html2canvas.js and jsPDF, or Print.js?😪
